relay-server/src/actors/healthcheck.rs

```diff
@@ -41,27 +58,39 @@ impl fmt::Display for SendError {
 
 impl Error for SendError {}
 
-/// Channel for sending public messages into a service.
+/// The address for a [`Service`].
 ///
-/// To send a message, use [`Addr::send`].
-#[derive(Clone, Debug)]
-pub struct Addr<T> {
-    tx: mpsc::UnboundedSender<Message<T>>,
+/// The address of a [`Service`] allows you to [send](Addr::send) messages to the service as
+/// long as the service is running.  It can be freely cloned.
+#[derive(Debug)]
+pub struct Addr<S: Service> {
+    tx: mpsc::UnboundedSender<S::Envelope>,
 }
 
-impl<T> Addr<T> {
+// For some reason the derive macro can't cope with this ¯\_(ツ)_/¯
```

Comment:
The reason is the `S` type parameter. The `Clone` derive generates (see [docs](https://doc.rust-lang.org/stable/std/clone/trait.Clone.html#derivable)):

```
impl<S: Clone> Clone for Addr<S> {}
```

It's perfectly fine and expected to implement `Clone` manually in such cases.